ios/testmanagerd/xctestrunutils.go

```diff
@@ -0,0 +1,173 @@
+package testmanagerd
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"github.com/danielpaulus/go-ios/ios"
+	"howett.net/plist"
+	"io"
+	"maps"
+	"os"
+	"path/filepath"
+)
+
+// XCTestRunData represents the structure of an .xctestrun file
+type XCTestRunData struct {
+	TestConfig        SchemeData        `plist:"-"`
+	XCTestRunMetadata XCTestRunMetadata `plist:"__xctestrun_metadata__"`
+}
+
+func (data XCTestRunData) toTestConfig(device ios.DeviceEntry, listener *TestListener) (TestConfig, error) {
+	testsToRun := data.TestConfig.OnlyTestIdentifiers
+	testsToSkip := data.TestConfig.SkipTestIdentifiers
+
+	testEnv := make(map[string]any)
+	maps.Copy(testEnv, data.TestConfig.EnvironmentVariables)
+	maps.Copy(testEnv, data.TestConfig.TestingEnvironmentVariables)
+
+	// Extract only the file name
+	var testBundlePath = filepath.Base(data.TestConfig.TestBundlePath)
+
+	// Build the TestConfig object from parsed data
+	testConfig := TestConfig{
+		TestRunnerBundleId: data.TestConfig.TestHostBundleIdentifier,
+		XctestConfigName:   testBundlePath,
+		Args:               data.TestConfig.CommandLineArguments,
+		Env:                testEnv,
+		TestsToRun:         testsToRun,
+		TestsToSkip:        testsToSkip,
+		XcTest:             !data.TestConfig.IsUITestBundle,
+		Device:             device,
+		Listener:           listener,
+	}
+
+	return testConfig, nil
+}
+
+// SchemeData represents the structure of a scheme-specific test configuration
+type SchemeData struct {
+	TestHostBundleIdentifier    string         `plist:"TestHostBundleIdentifier"`
+	TestBundlePath              string         `plist:"TestBundlePath"`
+	SkipTestIdentifiers         []string       `plist:"SkipTestIdentifiers"`
+	OnlyTestIdentifiers         []string       `plist:"OnlyTestIdentifiers"`
+	IsUITestBundle              bool           `plist:"IsUITestBundle"`
+	CommandLineArguments        []string       `plist:"CommandLineArguments"`
+	EnvironmentVariables        map[string]any `plist:"EnvironmentVariables"`
+	TestingEnvironmentVariables map[string]any `plist:"TestingEnvironmentVariables"`
+}
+
+// XCTestRunMetadata contains metadata about the .xctestrun file
+type XCTestRunMetadata struct {
+	FormatVersion int `plist:"FormatVersion"`
+}
+
+// XCTestRunCodec is a utility for parsing .xctestrun files with FormatVersion 1.
+// It extracts test configurations and metadata, providing a structured SchemeData object
+// that contains all the necessary information required to execute a test.
+// This includes details like test bundle paths, environment variables, command-line arguments,
+// and other configuration settings essential for running tests.
+
+// XCTestRunCodec handles encoding and decoding operations for .xctestrun files
+type XCTestRunCodec struct{}
```

Comment:
What's the purpose of this type? We can just declare the methods as functions (and since they are only used inside this package, we can unexport them as well)

Reply:
removed this and I did unexport the the methods so they are available only inside this package